build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -731,21 +731,31 @@ message ExecutedActionMetadata {
 
 // An ActionResult represents the result of an
 // [Action][build.bazel.remote.execution.v2.Action] being run.
+//
+// The server MUST NOT make any inlining decisions on his own but should
+// do so when requested by the client. The total serialized message size
+// SHOULD NOT exceed 1 MiB.
 message ActionResult {
   reserved 1; // Reserved for use as the resource name.
 
   // The output files of the action. For each output file requested in the
   // `output_files` field of the Action, if the corresponding file existed after
   // the action completed, a single entry will be present either in this field,
-  // or in the output_file_symlinks field, if the file was a symbolic link to
-  // another file.
+  // or the `inlined_output_files` field or the `output_file_symlinks` field if the
+  // file was a symbolic link to another file.
   //
   // If the action does not produce the requested output, or produces a
   // directory where a regular file is expected or vice versa, then that output
   // will be omitted from the list. The server is free to arrange the output
   // list as desired; clients MUST NOT assume that the output list is sorted.
   repeated OutputFile output_files = 2;
 
+  // The output files of the action whose contents are inlined into the request.
+  // A server MUST NOT inline outputs unless requested by the client in the
+  // [GetActionResultRequest][build.bazel.remote.execution.v2.GetActionResultRequest]
```

Comment:
MUST NOT->SHOULD NOT?

I'm hesitant to make it a MUST, in that it'll be mostly benign if it happens - no correctness guarantees will be violated due to over-inlining, just some amount of bandwidth wastage. As such, SHOULD NOT seems more correct (from the descriptions in https://www.ietf.org/rfc/rfc2119.txt).

Same below.

Reply:
Done, also below.